src/pipelines/token_classification.rs

```diff
@@ -0,0 +1,518 @@
+use crate::roberta::{RobertaForTokenClassification};
+use crate::distilbert::{DistilBertForTokenClassification, DistilBertConfig};
+use crate::Config;
+use crate::common::resources::{Resource, RemoteResource, download_resource};
+use serde::{Serialize,Deserialize};
+
+
+#[derive(Debug,Serialize,Deserialize)]
+/// # Token generated by a `TokenClassificationModel`
+pub struct Token {
+    /// String representation of the Token
+    pub text: String,
+
+    /// Confidence score
+    pub score: f64,
+
+    /// Token label (e.g. ORG, LOC in case of NER)
+    pub label: String,
+
+    /// Sentence index
+    #[serde(default)]
+    pub sentence: usize,
+
+    /// Token position index
+    pub index: u16,
+
+    /// Character offset (relative to begin of sentence)
+    pub char_offset: Option<u16>,
+
+    /// Word index
+    pub word_index: Option<u8>,
+
+    /// Continuation marker: marks this token as a continuation of the previous one
+    #[serde(default)]
+    pub continuation: bool,
+
+    /// Is this token followed by a space (false) or not (true) ?
+    #[serde(default)]
+    pub space: bool
+}
+
+/// # Configuration for TokenClassificationModel
+/// Contains information regarding the model to load and device to place the model on.
+pub struct TokenClassificationConfig {
+    /// Model type
+    pub model_type: ModelType,
+    /// Model weights resource (default: pretrained BERT model on CoNLL)
+    pub model_resource: Resource,
+    /// Config resource (default: pretrained BERT model on CoNLL)
+    pub config_resource: Resource,
+    /// Vocab resource (default: pretrained BERT model on CoNLL)
+    pub vocab_resource: Resource,
+    /// Merges resource (default: pretrained BERT model on CoNLL)
+    pub merges_resource: Option<Resource>,
+    /// Device to place the model on (default: CUDA/GPU when available)
+    pub device: Device,
+}
+
+impl TokenClassificationConfig {
+    /// Instantiate a new token classification configuration of the supplied type.
+    ///
+    /// # Arguments
+    ///
+    /// * `model_type` - `ModelType` indicating the model type to load (must match with the actual data to be loaded!)
+    /// * model - A `(local name, remote url)` tuple `(&str,&str)` pointing to the model to load (e.g.  model.ot)
+    /// * config - A `(local name, remote url)` tuple `(&str,&str)` pointing to the model configuration to load (e.g. config.json)
+    /// * vocab - A `(local name, remote url)` tuple `(&str,&str)` pointing to the tokenizer's vocabulary to load (e.g.  vocab.txt/vocab.json)
+    /// * vocab - An optional `(local name, remote url)` tuple (`Option<&str,str>`) pointing to the tokenizer's merge file to load (e.g.  merges.txt), needed only for Roberta.
+    ///
+    pub fn new(model_type: ModelType, model: (&str,&str), config: (&str,&str), vocab: (&str,&str), merges: Option<(&str,&str)>) -> TokenClassificationConfig {
```

Comment:
Could we keep `Resources` as an input here? The advantage is that the code would then work for both local and remote resources, as the rest of the code would run for both (`download_resource` is made to handle both and always return the  local resource path)

Reply:
Yes, that makes sense. I was kind of looking for a way to quickly get the ``Resource`` from a ``&str`` but I combining the two steps into one might indeed not be the best solution.